### ISSUE TITLE
Unable to login using registered account

### ISSUE THREAD

**Issue by FabioS82** [TwinProduction/PythonXatBot#1]
Good afternoon, when I try to use a xat account to log in to the bot it error. :( Anyone could see what was going please?

![screenshot_2](https://cloud.githubusercontent.com/assets/18095638/17644102/53512df6-6175-11e6-8abe-bac7d1f588dd.jpg)

**Comment by TwiN** [comment]
Try to use it without using a xat account for the mean time, I believe that Xat changed a couple of stuffs with their login packets and I'll need to fix that when I have time.

**Comment by FabioS82** [comment]
Yes, it works perfectly without login. I have fun me to edit it to use in the room where use. I await anxiously for the resolution of the bug to be able to get new things in the bot and then share.

I leave here one more observation:

```
   if DEBUG: print "Key '"+str(key)+"' was not found, thus it cannot be deleted."
```

this line causes an error, and put the bot down
thanks for sharing the bot friend!

**Comment by grimacecoin** [comment]
Still this error continues and still is not repaired but this bot is not bad, another thing is that the bot does not last more than 5 minutes in a room because it is disconnected.

**Comment by TwiN** [comment]
@xYoiker I just tried to replicate that and on my end, it stays longer than 5 minutes without any problem. However, I have an idea as to why this might be happening to you.

Since the introduction of the (BOT) power, xat has made it more difficult for bots, especially when they're guests, to be chatting at the same time as their owner.
For instance, if I were to go to a chat as a guest and bring the bot in as a guest too, it is very likely that the bot would disconnect. The same situation can be observered when a guest spams too quickly, he automatically disconnects.

The only solution I can tell you really is, try to make it so both you and your bot are at least members rather than guests - at least on my end, that has always worked.

If that is not an option, then implementing a proxy or running the bot on a VPS would also due the trick because the packets comes from different different IPs.

**Comment by grimacecoin** [comment]
I understand, one question there will be no more updates for this bot?

**Comment by TwiN** [comment]
@xYoiker Not for the meantime, I'm busy with work and moving out lately. This was just a basic bot I released for fun to begin with, not any serious project. 

However, when I have some free time, I'll see if I can fix a couple of things on it. As for when I'll have time for it, I have no idea

**Comment by merkader69** [comment]
I noticed they've added L4, L3, as well as L2 parameters to this line: ``` gSocket.sendall('<j2 cb="0" l5="'+pkt_l5+'" y="'+pkt_i+'" k="'+botK1+'" p="0" c="'+str(chatid)+'" f="2" u="'+str(botID)+'" d0="0" n="'+botDisplayName+'" a="997" h="" v="10" />\x00')```, where as obviously the old one only holds the L5, i'm a novice at best, so would you happen to be able to point me in the direction of something able to generate said layer values that they've added until your return?
